test/proposal.test.ts

```diff
@@ -306,3 +323,167 @@ describe("proposal", function () {
+    threshold: 2,
+  };
+  const virtualTokenCreationSettings: VirtualTokenCreationSettings = {
+    merkleRoot: "0x" + "42".repeat(32),
+    usdcToken: "0x0000" + "42".repeat(17) + "01",
+    gnoToken: "0x0000" + "42".repeat(17) + "02",
+    gnoPrice: "31337",
+    wrappedNativeToken: "0x0000" + "42".repeat(17) + "03",
+    nativeTokenPrice: "42424242",
+  };
+
+  let settings: DeploymentProposalSettings;
+  describe("relay of safe deployment", function () {
+    let gnosisSafeDefaults: SafeDeploymentAddresses;
+    before(async function () {
+      const bridgeParameters: BridgeParameter = {
+        multiTokenMediatorGnosisChain: "0x" + "01".repeat(20),
+        multiTokenMediatorETH: "0x" + "02".repeat(20),
+        arbitraryMessageBridgeETH: arbitraryMessageBridge.address,
+      };
+      settings = {
+        gnosisDao: gnosisDao.address,
+        cowDao: cowDaoSettings,
+        teamController: teamConrollerSettings,
+        cowToken: {},
+        virtualCowToken: virtualTokenCreationSettings,
+        bridge: bridgeParameters,
+      };
+      gnosisSafeDefaults = gnosisSafeManager.getDeploymentAddresses();
+    });
+    const expectedCowDaoAddress = "0x6a54ef9C6BE1aF4099336C3bDBBDf690d0B67A7c";
+    it("has the correct target address", async function () {
+      const bridgedGnosisSafeDeployment = await createTxForBridgedSafeSetup(
+        expectedCowDaoAddress,
+        {
+          arbitraryMessageBridgeETH: settings.bridge.arbitraryMessageBridgeETH,
+        },
+        settings.cowDao,
+        gnosisSafeDefaults,
+        hre.ethers,
+      );
+      expect(bridgedGnosisSafeDeployment.to).to.be.equal(
+        settings.bridge.arbitraryMessageBridgeETH,
+      );
+    });
+    it("createTxForBridgedSafeSetup calls the bridge correctly on requireToPassMessage", async function () {
+      const bridgedGnosisSafeDeployment = await createTxForBridgedSafeSetup(
+        expectedCowDaoAddress,
+        {
+          arbitraryMessageBridgeETH: settings.bridge.arbitraryMessageBridgeETH,
+        },
+        settings.cowDao,
+        gnosisSafeDefaults,
+        hre.ethers,
+      );
+      const functionSignatureBytes = 4;
+      const [to, data, gasLimit] = defaultAbiCoder.decode(
+        ["address", "bytes", "uint256"],
+        utils
+          .arrayify(bridgedGnosisSafeDeployment.data)
+          .slice(functionSignatureBytes),
+      );
+      await arbitraryMessageBridge.mock.requireToPassMessage
+        .withArgs(to, data, gasLimit)
+        .reverts();
+      await expect(
+        execSafeTransaction(gnosisDao, bridgedGnosisSafeDeployment, [
+          gnosisDaoOwner,
+        ]),
+      ).to.be.reverted;
+      await arbitraryMessageBridge.mock.requireToPassMessage
+        .withArgs(to, data, gasLimit)
+        .returns(messageID);
+      await expect(
+        execSafeTransaction(gnosisDao, bridgedGnosisSafeDeployment, [
+          gnosisDaoOwner,
+        ]),
+      ).to.not.be.reverted;
```

Comment:
Nice! This resolves my [comment from here](https://github.com/gnosis/cow-token/pull/72/files#r792486292).